pkg/adapter/kubernetesevents/adapter.go

```diff
@@ -98,16 +101,16 @@ func (a *Adapter) addEvent(new interface{}) {
 	logger := logging.FromContext(context.TODO()).With(zap.Any("eventID", event.ObjectMeta.UID), zap.Any("sink", a.SinkURI))
 	logger.Debug("GOT EVENT", event)
 
+	var err error
+	a.initClientOnce.Do(func() {
```

Comment:
Just curious about doing this instead of creating this in Start() method like where you create the kubeClient.